src/cobra/core/model.py

```diff
@@ -676,7 +676,8 @@ def add_boundary(
                 "identifier. Please set the `reaction_id`."
             )
         if reaction_id in self.reactions:
-            raise ValueError(f"Boundary reaction '{reaction_id}' already exists.")
+            # It already exists so just retrieve it.
```

Comment:
Maybe add a log message that this happened? Does not have to be warning, but an info might be good.